algolia/transport/transport.go

```diff
@@ -193,7 +224,21 @@ func buildRequest(
 		if err != nil {
 			return nil, fmt.Errorf("cannot serialize request body:\n\terr=%v\n\tmethod=%s\n\turl=%s\n\tbody=%#v", err, method, urlStr, body)
 		}
-		req, err = http.NewRequest(method, urlStr, bytes.NewReader(data))
+
+		var r io.ReadCloser
+
+		if shouldCompress(c, method) {
+			switch c {
+			case compression.GZIP:
+				r = gzipCompressionReadCloser(data)
+			default:
+				r = noCompressionReadCloser(data)
+			}
+			headers["Content-Encoding"] = "gzip"
```

Comment:
Is the standard http client setting the `Content-Lenght` header as well?

Reply:
Yes indeed. As per the documentation of `http.Request.Header`:
```
// For client requests, certain headers such as Content-Length
// and Connection are automatically written when needed and
// values in Header may be ignored. See the documentation
// for the Request.Write method.
```